### DISCUSSION TITLE
How to change / set the SELECT default value?

### DISCUSSION THREAD

**Discussion by NgYueHong** [filamentphp/filament#13035] [Help]
### Package

Panel builder

### Package Version

3.2

### How can we help you?

I am trying to set the `select` value to be `credit` when `$get('type') == "credit"`. I have tried to put the code `$set('payment_method','Credit');` on `hidden()`, `disabled()`, but all of it were executed before `options()`. How can I set the `select` value to be `credit`? Where should I put the code so that it will be executed after `options()`?

```
Select::make('payment_method')
    ->required()
    ->hidden(function($get,$set){
        if ($get('type') == "credit") {
            $set('payment_method','Credit');
        }
        return false;
    })
    ->options(function ($get) {
        if ($get('type') == "cash") {
            return ['Cash' => 'Cash', 'AliPay' => 'AliPay'];
        } else {
            return ['Credit' => 'Credit'];
        }
    })
```

**Comment by leandrocfe** [comment]
You should do this https://filamentphp.com/docs/3.x/forms/advanced#dependant-select-options

**Reply by NgYueHong** [reply]
Thanks a lot for your help. But how can I get the 'payment_method' select field to select 'Credit' as the value?
